src/ParallelAlgorithms/Amr/Actions/CreateChild.hpp

```diff
@@ -50,6 +51,12 @@ struct CreateChild {
       const size_t index_of_child_id,
       const std::unordered_map<Parallel::Phase, size_t>
           parent_phase_bookmarks) {
+    if (CHARM_VERSION_MAJOR < 8 and
+        Parallel::number_of_procs<size_t>(cache) > 1) {
+      ERROR_NO_TRACE(
+          "Dynamically creating elements in parallel executables is broken "
+          "until charm 8.");
```

Comment:
It's not quite this bad, is it? I've been using AMR with h-refinement in a separate phase in elliptic executables for a while now with no issues.

Reply:
I believe it occurs when one of the initial elements is destroyed and then recreated in a later iteration on a PE other than PE 0.   If the current elliptic code usage only refines, and never coarsens, this condition won't occur.

See https://github.com/kidder/charm_spectre_tests/tree/main/dynamic_array_hang

Reply:
okay, then probably only erroring in CreateParent would be adequate